core/src/main/java/hudson/model/AbstractCIBase.java

```diff
@@ -160,37 +163,40 @@ private void updateComputer(Node n, Map<String,Computer> byNameMap, Set<Computer
      * This method tries to reuse existing {@link Computer} objects
      * so that we won't upset {@link Executor}s running in it.
      */
-    protected void updateComputerList(boolean automaticSlaveLaunch) throws IOException {
```

Comment:
While we are here, consider fixing `Jenkins.setNumExecutors` to call this. Last I checked you had to follow that up with `j.setNodes(j.getNodes())`, which seems ridiculous.